### ISSUE TITLE
Stack overflow when asking for a password with less length than different charsets

### ISSUE THREAD

**Issue by dislick** [dislick/ts-pwgen#3]
## Cause
The password generator method will recursively call itself if the generated password does not contain at least one character from each charset. It uses four charsets per default, so trying to create a password of length 3 will result in a stack overflow.

## Reproduce
`pwgen -l 3`
```
(node:85475) UnhandledPromiseRejectionWarning: Unhandled promise rejection (rejection id: 1): RangeError: Maximum call stack size exceeded
(node:85475) DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
```

## Fix Proposal
This error is very easy to catch, simply check the lengths of the charsets and the password and tell the user that this will not work.

**Comment by dislick** [comment]
Fixed with pull request #4

**Comment by DevTrong** [comment]
![image](https://cloud.githubusercontent.com/assets/3624351/21586470/c0fb796c-d0d1-11e6-9769-c902f247466f.png)

Should take parts into consideration.
